### ISSUE TITLE
Incorrect text clipping in presence of multiple subplots

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#6794]
This is a regression in 2.0b3 compared to 1.5.2 (haven't bisected more)
```
from matplotlib import pyplot as plt
import numpy as np

np.random.seed(1)
fig, (ax1, ax2) = plt.subplots(2)
for x, y in np.random.rand(10, 2):
    ax1.text(x, y, "foo", clip_on=True)
    ax2.text(x, y, "foo")
plt.show()
```
1.5.2 displays everything fine
![152](https://cloud.githubusercontent.com/assets/1322974/16928190/33e8555a-4ce6-11e6-9739-0fd3851065b0.png)
2.0b3 clips away all text objects in the top axes
![20b3](https://cloud.githubusercontent.com/assets/1322974/16928200/41660a10-4ce6-11e6-9410-ea4a956c95bb.png)
(so I'm marking this as 2.0 but feel free to change it)

The texts actually appear to be clipped to the "wrong" place and use the wrong zorder: you can see them appear "behind the axes" if you pan the axes as below:
![panned](https://cloud.githubusercontent.com/assets/1322974/16928282/a353163c-4ce6-11e6-9964-126eaa608f17.png)

**Comment by tacaswell** [comment]
yikes!

**Comment by tacaswell** [comment]
I can confirm this and agree it needs to be bisected.  Nothing obvious is jumping out at me.

**Comment by jenshnielsen** [comment]
I have bisected it and it looks like the first bad commit is the merge of  #5361 

```
a340827407cf8397f2aea4a94fba2dbf2691e217 is the first bad commit
commit a340827407cf8397f2aea4a94fba2dbf2691e217
Author: Thomas A Caswell <tcaswell@gmail.com>
Date:   Wed Nov 4 22:57:46 2015 -0500

    Merge pull request #5361 from mdboom/fast-text

    Fix #347: Faster text rendering in Agg

:040000 040000 318ca4105c13e269f0258fb2fd833ba0a804c94c ab984a41574b5f97361c4b2843d6f0d166dfa2be M	doc
:040000 040000 7b0d1be25da84428a1f661c895f6122ae5a842db 9c561f043224b6155097d0ac469ca5df689b2d5d M	src
```

**Comment by jenshnielsen** [comment]
I can also confirm that neither the cairo or pdf backend has the issue.
